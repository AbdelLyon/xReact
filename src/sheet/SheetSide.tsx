import { PropsWithChildren, ReactNode } from "react";

import {
  Sheet,
  SheetContent,
  SheetTitle,
  SheetTrigger,
  SheetClose,
} from "@/ui/sheet";

type Props = PropsWithChildren<{
  classNames?: {
    content?: string;
    trigger?: string;
  };
  trigger: ReactNode;
  title?: string;
  direction?: "left" | "top" | "bottom" | "right";
  open?: boolean;
  onOpenChange?: (value: boolean) => void;
}>;

const SheetSide = ({
  trigger,
  classNames,
  children,
  title,
  direction = "right",
  open,
  onOpenChange,
}: Props) => {
  return (
    <Sheet open={open} onOpenChange={onOpenChange}>
      <SheetTrigger asChild className={classNames?.trigger}>
        {trigger}
      </SheetTrigger>
      <SheetContent className={classNames?.content} side={direction}>
        {title && <SheetTitle className=" absolute top-2">{title}</SheetTitle>}
        <div className="mt-4">{children}</div>
      </SheetContent>
    </Sheet>
  );
};

export { SheetSide, SheetClose };
