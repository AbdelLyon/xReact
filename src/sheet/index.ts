import "@/index.css";
export { SheetSide, SheetClose } from "@/sheet/SheetSide";
