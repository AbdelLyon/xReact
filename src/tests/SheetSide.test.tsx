import { describe } from "node:test";

import { render } from "@testing-library/react";
import { expect, it } from "vitest";

import { SheetSide } from "@/sheet";

describe("SheetSide Component", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(
      <SheetSide
        classNames={{ content: "custom-content", trigger: "custom-trigger" }}
        direction="right"
        title="Test Title"
        trigger={<button>Open</button>}
      >
        <div>Content of the sheet</div>
      </SheetSide>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
