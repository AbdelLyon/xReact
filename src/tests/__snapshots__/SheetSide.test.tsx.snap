// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <button
    aria-controls="radix-:r0:"
    aria-expanded="false"
    aria-haspopup="dialog"
    class="custom-trigger"
    data-state="closed"
    type="button"
  >
    Open
  </button>
</DocumentFragment>
`;
